lua/copilot/util.lua

```diff
@@ -230,6 +230,45 @@ function M.get_editor_configuration()
   }
 end
 
+---@param str string
+local function url_decode(str)
+  return vim.fn.substitute(str, [[%\(\x\x\)]], [[\=iconv(nr2char("0x".submatch(1)), "utf-8", "latin1")]], "g")
+end
+
+---@return copilot_network_proxy|nil
+function M.get_network_proxy()
+  local proxy_uri = vim.g.copilot_proxy
```

Comment:
Shouldn't this work?
```lua
local proxy_uri = vim.g.copilot_proxy or vim.env.https_proxy or vim.env.http_proxy
```

Reply:
Environment variables are case sensitive. Usually those variables are named `HTTP_PROXY` and `HTTPS_PROXY`. And also there's another `NO_PROXY` variable usually associated with them, that defines which urls should bypass the proxy. So if we support proxy using environment variable we "should" support all of those variables, otherwise it would be incomplete.

I'd rather not go down that road right now...

You can simply do

```lua
vim.g.copilot_proxy = vim.env.HTTPS_PROXY or vim.env.HTTP_PROXY
```

in your config file to get the same effect.

Reply:
ok will do thanks.